ir_api/core/model.py

```diff
@@ -0,0 +1,132 @@
+"""
+This module contains classes and enums that define the structure of the database tables and relationships for storing
+information about instruments, runs, scripts, and reductions.
+"""
+from __future__ import annotations
+
+import enum
+from datetime import datetime
+from typing import Optional, List
+
+from sqlalchemy import Table, Column, ForeignKey, String, DateTime, Enum, Integer, inspect
+from sqlalchemy.dialects.postgresql import JSONB
+from sqlalchemy.orm import Mapped, mapped_column, relationship, DeclarativeBase
+
+
+class ReductionState(enum.Enum):
```

Comment:
You should probably use the SQLAlchemy enum in this one (It inherits like 7 classes so you need a pylint ignore):
```suggestion
class ReductionState(Enum):  # pylint: disable=too-many-ancestors
```

Reply:
The enum type is mapped to sqlalchemy enum automatically in the model definition.
see: https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html#using-python-enum-or-pep-586-literal-types-in-the-type-map